pkg/security/auth.go

```diff
@@ -140,6 +140,17 @@ func UserAuthCertHook(insecureMode bool, tlsState *tls.ConnectionState) (UserAut
 			return nil, nil
 		}
 
+		// The client certificate should not be a tenant client type. For now just
+		// check that it doesn't have OU=Tenants. It would make sense to add
+		// explicit OU=Users to all client certificates and to check for match.
+		ous := tlsState.PeerCertificates[0].Subject.OrganizationalUnit
+		for _, ou := range ous {
+			if ou == tenantsOU {
+				return nil,
+					errors.Errorf("using tenant client certificate as user certificate is not allowed")
```

Comment:
Is there a reasonably easy way to test this, for example in `TestAuthHook`?

Reply:
The test that I added in `TestAuthenticationHook` is testing exactly this. It shows that certificate for CN=foo with OU=Tenants can not be used to authenticate user foo
```
		// Secure mode, tenant cert, foo user.
		{false, "(Tenants)foo", "foo", "", true, false, false},
```